core/templates/pages/exploration-editor-page/exploration-editor-page.component.spec.ts

```diff
@@ -493,6 +501,54 @@ describe('Exploration editor page component', function() {
     });
   });
 
+  describe('Checking internet Connection', () => {
+    beforeEach(() => {
+      registerAcceptTutorialModalEventSpy = (
+        spyOn(sas, 'registerAcceptTutorialModalEvent'));
+      registerDeclineTutorialModalEventSpy = (
+        spyOn(sas, 'registerDeclineTutorialModalEvent'));
+      spyOn(cs, 'getExplorationId').and.returnValue(explorationId);
+      spyOn(efbas, 'fetchExplorationFeaturesAsync')
+        .and.returnValue($q.resolve({}));
+      spyOn(eis, 'initAsync').and.returnValue(Promise.resolve());
+      spyOn(eis, 'flushUpdatedTasksToBackend')
+        .and.returnValue(Promise.resolve());
+      spyOn(ews, 'updateWarnings').and.callThrough();
+      spyOn(gds, 'recompute').and.callThrough();
+      spyOn(pts, 'setPageTitle').and.callThrough();
+      spyOn(tds, 'getOpenThreadsCountAsync').and.returnValue($q.resolve(0));
+      spyOn(ueps, 'getPermissionsAsync')
+        .and.returnValue($q.resolve({canEdit: true, canVoiceover: true}));
+      spyOnProperty(stfts, 'onOpenEditorTutorial').and.returnValue(
+        mockOpenEditorTutorialEmitter);
+      spyOnProperty(cns, 'onInternetStateChange').and.returnValue(
+        mockConnectionServiceEmitter);
+      spyOnProperty(stfts, 'onOpenTranslationTutorial').and.returnValue(
+        mockOpenTranslationTutorialEmitter);
+      spyOn(as, 'addInfoMessage');
+      spyOn(as, 'addSuccessMessage');
+      explorationData.is_version_of_draft_valid = false;
```

Comment:
Do we need all these spyon here?

Reply:
Yes, Inorder to spy all the other services going to run when we call this init function.